src/middleware/fetchRequest.js

```diff
@@ -4,6 +4,7 @@ import onRejection from '../utils/onRejection';
 
 const fetchRequest = (store, next, action) => {
   const req = container.getDefinition('requestBuilder').getArgument('build')(action.request.url, {
+    ...action.request,
```

Comment:
If you pass the whole request object here, separately setting the body and headers (r9 and r10) can be removed.